### ISSUE TITLE
False positive from Warning.UnusedStringOperation when result of operation is passed to a function on a dynamic module

### ISSUE THREAD

**Issue by kerryb** [rrrene/credo#974]
### Precheck

* Proposals for new features should be submitted via: https://github.com/rrrene/credo-proposals
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`):
  - 1.6.4-ref.main.6767ee049+uncommittedchanges
* Erlang/Elixir version (`elixir -v`):
  - Erlang/OTP 25 [erts-13.0] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit:ns]
  - Elixir 1.13.4 (compiled with Erlang/OTP 25)
* Operating system:
  - macOS 12.4 (21F79) 

### What were you trying to do?

Run `mix credo` on the following code (this is the most I’ve managed to reduce the actual code down and still demonstrate the issue):

```elixir
defmodule Foo do
  def foo(module, value) do
    module.fun(String.to_integer(value))
    :ok
  end
end

defmodule Bar do
  def fun(arg) do
    IO.inspect(arg)
  end
end

# Foo.foo(Bar, "42")

```

### Expected outcome

No `Warning.UnusedStringOperation` warning

### Actual outcome

```
  Warnings - please take a look
┃
┃ [W] ↗ There should be no unused return values for String functions.
┃       lib/foo.ex:3:16 #(Foo.foo)
```

If I get time later I’ll try to delve into the Credo code and see whether I can figure out a fix, in which case I’ll submit a PR.

**Comment by rrrene** [comment]
Thx for reporting this! :+1: It should be be fixed in the latest version of Credo.

If it is not, please feel free to re-open this issue!
